CMakeLists.txt

```diff
@@ -128,58 +178,204 @@ set(IO_SOURCE
-    target_link_libraries(poc_HDF5Writer ${Boost_LIBRARIES})
-    target_link_libraries(poc_HDF5Reader ${Boost_LIBRARIES})
-endif ()
-if (LIBOPENPMD_PARALLEL)
-    target_link_libraries(libopenpmdAuxiliaryTests ${MPI_LIBRARIES})
-    target_link_libraries(libopenpmdCoreTests ${MPI_LIBRARIES})
-    target_link_libraries(libopenpmdSerialIOTests ${MPI_LIBRARIES})
-    target_link_libraries(libopenpmdParallelIOTests ${MPI_LIBRARIES})
-    target_link_libraries(poc_HDF5Writer ${MPI_LIBRARIES})
-    target_link_libraries(poc_HDF5Reader ${MPI_LIBRARIES})
-endif ()
+
+# library
+add_library(openPMD.core ${CORE_SOURCE})
+add_library(openPMD.io ${IO_SOURCE})
+
+# own headers
+target_include_directories(openPMD.core PUBLIC
+    $<BUILD_INTERFACE:${openPMD_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${openPMD_BINARY_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+target_include_directories(openPMD.io PUBLIC
+    $<BUILD_INTERFACE:${openPMD_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${openPMD_BINARY_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+
+if(TARGET Boost::filesystem)
+    target_link_libraries(openPMD.core PUBLIC
+        Boost::boost Boost::system Boost::filesystem)
+    target_link_libraries(openPMD.io PUBLIC
+        Boost::boost Boost::system Boost::filesystem)
+else()
+    target_link_libraries(openPMD.core PUBLIC
+        ${Boost_LIBRARIES})
+    target_link_libraries(openPMD.io PUBLIC
+        ${Boost_LIBRARIES})
+    target_include_directories(openPMD.core SYSTEM PUBLIC ${Boost_INCLUDE_DIRS})
+
+    target_include_directories(openPMD.io SYSTEM PUBLIC ${Boost_INCLUDE_DIRS})
+endif()
+
+if(openPMD_HAVE_MPI)
+    # MPI targets: CMake 3.9+
+    # note: often the PUBLIC dependency to CXX is missing in C targets...
+    target_link_libraries(openPMD.core PUBLIC MPI::MPI_C MPI::MPI_CXX)
+    target_link_libraries(openPMD.io PUBLIC MPI::MPI_C MPI::MPI_CXX)
+
+    target_compile_definitions(openPMD.core PUBLIC "-DLIBOPENPMD_WITH_MPI=ON")
+    target_compile_definitions(openPMD.io PUBLIC "-DLIBOPENPMD_WITH_MPI=ON")
+else()
+    target_compile_definitions(openPMD.core PUBLIC "-D_NOMPI=ON")
+    target_compile_definitions(openPMD.io PUBLIC "-D_NOMPI=ON")
+endif()
+
+if(openPMD_HAVE_HDF5)
+    target_link_libraries(openPMD.io PUBLIC ${HDF5_LIBRARIES})
+    target_include_directories(openPMD.io SYSTEM PUBLIC ${HDF5_INCLUDE_DIRS})
+    target_compile_definitions(openPMD.io PUBLIC ${HDF5_DEFINITIONS})
+    target_compile_definitions(openPMD.io PRIVATE "-DLIBOPENPMD_WITH_HDF5=ON")
+    if(openPMD_HAVE_MPI)
+        # TODO: remove and just rely on "WITH_MPI"
+        target_compile_definitions(openPMD.io PUBLIC "-DLIBOPENPMD_WITH_PARALLEL_HDF5=ON")
+    endif()
+endif()
+
+if(openPMD_HAVE_ADIOS1)
+    target_link_libraries(openPMD.io ${ADIOS_LIBRARIES})
+    target_include_directories(openPMD.io SYSTEM PUBLIC ${ADIOS_INCLUDE_DIRS})
+    target_compile_definitions(openPMD.io PRIVATE "-DLIBOPENPMD_WITH_ADIOS1=ON")
```

Comment:
All compile definitions LIBOPENPMD_WITH_IOBACKEND also have to be visible in serial IO tests.